cmd/appgw-ingress/main.go

```diff
@@ -74,23 +72,48 @@ func main() {
 		version.PrintVersionAndExit()
 	}
 
-	if err := environment.ValidateEnv(env); err != nil {
-		glog.Fatal("Error while initializing values from environment. Please check helm configuration for missing values.", err)
-	}
-
 	// Workaround for "ERROR: logging before flag.Parse"
 	// See: https://github.com/kubernetes/kubernetes/issues/17162#issuecomment-225596212
 	_ = flag.CommandLine.Parse([]string{})
 	_ = flag.Lookup("logtostderr").Value.Set("true")
 	_ = flag.Set("v", strconv.Itoa(*verbosity))
 
-	appGwClient := n.NewApplicationGatewaysClient(env.SubscriptionID)
+	apiConfig := getKubeClientConfig()
+	kubeClient := kubernetes.NewForConfigOrDie(apiConfig)
+	crdClient := versioned.NewForConfigOrDie(apiConfig)
+	istioCrdClient := istio.NewForConfigOrDie(apiConfig)
+	recorder := getEventRecorder(kubeClient)
+	namespaces := getNamespacesToWatch(env.WatchNamespace)
+	k8sContext := k8scontext.NewContext(kubeClient, crdClient, istioCrdClient, namespaces, *resyncPeriod)
+	agicPod := k8sContext.GetAGICPod(env)
+	metricStore := metricstore.NewMetricStore(env)
+
+	if env.AppGwName == "" {
+		env.AppGwName = env.ReleaseName
+	}
+
+	if subID, resGp, err := k8sContext.GetInfrastructureResourceGroupID(); err == nil {
+		env.SubscriptionID = string(subID)
+		env.ResourceGroupName = string(resGp)
+	}
+
+	//var azClient azure.AzClient := azure.NewAzClient(azure.SubscriptionID(env.SubscriptionID), azure.ResourceGroup(env.ResourceGroupName), azure.ResourceName(env.AppGwName), )
 	var err error
-	if appGwClient.Authorizer, err = getAuthorizerWithRetry(env, maxAuthRetryCount); err != nil {
+	var authorizer autorest.Authorizer
+	if authorizer, err = azure.GetAuthorizerWithRetry(env.AuthLocation, maxAuthRetryCount, retryPause); err != nil {
 		glog.Fatal("Failed obtaining authentication token for Azure Resource Manager")
 	}
-	if err = waitForAzureAuth(env, appGwClient, maxAuthRetryCount); err != nil {
-		glog.Fatal("Failed authenticating with Azure Resource Manager")
+
+	azClient := azure.NewAzClient(azure.SubscriptionID(env.SubscriptionID), azure.ResourceGroup(env.ResourceGroupName), azure.ResourceName(env.AppGwName), authorizer)
+	if err = azure.WaitForAzureAuth(azClient, maxAuthRetryCount, retryPause); err != nil {
```

Comment:
I wonder if we should add a flag of some sort here, which would indicate explicitly OK with AGIC creating an App Gateway.
I'm worried that once we release this - customers using AGIC in the current scenario way run into mis-configuration, where the App Gateway exists, but is not found, and this here creates it and increases the bill unnecessarily.  In that case this could/should have been a legitimate error.